File: API/ClientApp/src/app/models/RegisterDTO.ts
export class RegisterDTO {
    
    constructor() {
        this.firstname = "";
        this.lastname = "";
        this.email = "";
        this.password = "";
        this.passwordRepeat = "";
        this.birthdate = new Date();
        this.addressLine1 = "";
        this.addressLine2 = "";
        this.postCode = "";
    }

    firstname: string;
    lastname: string;
    email: string;
    password: string;
    passwordRepeat: string;
    birthdate: any;
    addressLine1: string;
    addressLine2: string;
    postCode: string;
}

File: API/ModelsDTO/RegisterDTO.cs
namespace API.ModelsDTO
{
    public class RegisterDTO
    {
        public string Firstname { get; set; }
        public string Lastname { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public string PasswordRepeat { get; set; }
        public DateTime Birthdate { get; set; }
        public string AddressLine1 { get; set; }
        public string AddressLine2 { get; set; }
        public string PostCode { get; set; }
    }
}
